  Recent versions of glib don't work inside an extern "C" block, but some seafile sources
  include searpc-client.h inside one. See https://github.com/haiwen/libsearpc/pull/57.

  This is the easy fix, needed until upstream fixes the problem.
--- a/lib/searpc-client.h~	2020-06-18 05:23:11.000000000 +0200
+++ b/lib/searpc-client.h	2021-04-10 15:48:09.994005146 +0200
@@ -7,8 +7,14 @@
 #define LIBSEARPC_API
 #endif
 
+#ifdef __cplusplus
+extern "C++" {
+#endif
 #include <glib.h>
 #include <glib-object.h>
+#ifdef __cplusplus
+}
+#endif
 #include <jansson.h>
 
 #ifndef DFT_DOMAIN
